CodeEdit/Settings/ExecutionSettingsView.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  ExecutionSettingsView.swift
+//  CodeEdit
+//
+//  Created by 朱浩宇 on 2022/3/23.
+//
+
+import SwiftUI
+import TerminalEmulator
+import FontPicker
+import Preferences
+
+struct ExecutionSettingsView: View {
+    @AppStorage(TerminalShellType.storageKey) var shellType: TerminalShellType = .default
+    @AppStorage(TerminalFont.storageKey) var terminalFontSelection: TerminalFont = .default
+    @AppStorage(TerminalFontName.storageKey) var terminalFontName: String = TerminalFontName.default
+    @AppStorage(TerminalFontSize.storageKey) var terminalFontSize: Int = TerminalFontSize.default
+
+    @State var customFont  = false
+
+    @StateObject private var colors = AnsiColors.shared
+    @State var terminalColor: Color = .gray
+
+    @Environment(\.colorScheme) var colorScheme
+
+    var body: some View {
+        VStack(alignment: .center) {
+            Preferences.Container(contentWidth: 500) {
+                Preferences.Section(title: "Terminal Shell:") {
+                    Picker("", selection: $shellType) {
+                        Text("System Default".localized())
+                            .tag(TerminalShellType.auto)
+                        Text("ZSH")
+                            .tag(TerminalShellType.zsh)
+                        Text("Bash")
+                            .tag(TerminalShellType.bash)
+                    }
+                    .fixedSize()
+                }
+
+                Preferences.Section(title: "Terminal Font:") {
+                    Toggle(isOn: $customFont) {
```

Comment:
Use a separate line for switching between `system` and custom font. With the font picker only shown when `custom` is selected. like on current `main`

Reply:
But I don't think this design will cause misunderstanding. And the content can be reduced.

Reply:
Having an unlabeled checkbox near a font selector will most likely lead to misunderstandings.